apps/desktop/electron/database/migrations/schema.ts

```diff
@@ -0,0 +1,197 @@
+        notes: text(),
+    },
+    (table) => [
+        index("index_marcher_pages_on_page_id").on(table.page_id),
+        index("index_marcher_pages_on_marcher_id").on(table.marcher_id),
+    ],
+);
+
+export const field_properties = sqliteTable("field_properties", {
+    id: integer().primaryKey(),
+    json_data: text().notNull(),
+    image: blob(),
+});
+
+export const audio_files = sqliteTable("audio_files", {
+    id: integer().primaryKey(),
+    path: text().notNull(),
+    nickname: text(),
+    data: blob(),
+    selected: integer().default(0).notNull(),
+    created_at: text().notNull(),
+    updated_at: text()
+        .notNull()
+        .$onUpdate(() => "sql`(CURRENT_TIMESTAMP)`"),
+});
+
+export const shapes = sqliteTable("shapes", {
+    id: integer().primaryKey(),
+    name: text(),
+    created_at: text().default("sql`(CURRENT_TIMESTAMP)`").notNull(),
+    updated_at: text()
+        .default("sql`(CURRENT_TIMESTAMP)`")
+        .notNull()
+        .$onUpdate(() => "sql`(CURRENT_TIMESTAMP)`"),
+    notes: text(),
+});
+
+export const shape_pages = sqliteTable("shape_pages", {
+    id: integer().primaryKey(),
+    shape_id: integer()
+        .notNull()
+        .references(() => shapes.id, { onDelete: "cascade" }),
+    page_id: integer()
+        .notNull()
+        .references(() => pages.id, { onDelete: "cascade" }),
+    svg_path: text().notNull(),
+    created_at: text().default("sql`(CURRENT_TIMESTAMP)`").notNull(),
+    updated_at: text()
+        .default("sql`(CURRENT_TIMESTAMP)`")
+        .notNull()
+        .$onUpdate(() => "sql`(CURRENT_TIMESTAMP)`"),
+    notes: text(),
+});
+
+export const shape_page_marchers = sqliteTable(
+    "shape_page_marchers",
+    {
+        id: integer().primaryKey(),
+        shape_page_id: integer()
+            .notNull()
+            .references(() => shape_pages.id),
+        marcher_id: integer()
+            .notNull()
+            .references(() => marchers.id),
+        position_order: integer(),
+        created_at: text().default("sql`(CURRENT_TIMESTAMP)`").notNull(),
+        updated_at: text()
+            .default("sql`(CURRENT_TIMESTAMP)`")
+            .notNull()
+            .$onUpdate(() => "sql`(CURRENT_TIMESTAMP)`"),
+        notes: text(),
+    },
+    (table) => [
+        index("idx-spm-marcher_id").on(table.marcher_id),
+        index("idx-spm-shape_page_id").on(table.shape_page_id),
+    ],
+);
+
+export const section_appearances = sqliteTable("section_appearances", {
+    id: integer().primaryKey(),
+    section: text().notNull(),
+    fill_color: text().default("rgba(0, 0, 0, 1)").notNull(),
+    outline_color: text().default("rgba(0, 0, 0, 1)").notNull(),
+    shape_type: text().default("circle").notNull(),
+    created_at: text().default("sql`(CURRENT_TIMESTAMP)`").notNull(),
+    updated_at: text()
+        .default("sql`(CURRENT_TIMESTAMP)`")
+        .notNull()
+        .$onUpdate(() => "sql`(CURRENT_TIMESTAMP)`"),
```

Comment:
This I added to all the tables. The $ means it is enforced on the js side but means we never have to think about setting updated_at in the app code.